specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,485 @@
+of the `"output_labels"` attribute attached to the entry point.
+
+## Module Flags Metadata
+
+The following [module
+flags](https://llvm.org/docs/LangRef.html#module-flags-metadata) must be added
+to the QIR bitcode:
+
+- a flag with the string identifier `"qir_major_version"` that contains a
+  constant value of type `i32`
+- a flag with the string identifier `"qir_minor_version"` that contains a
+  constant value of type `i32`
+- a flag with the string identifier `"dynamic_qubit_management"` that contains a
+  constant `true` or `false` value of type `i1`
+- a flag with the string identifier `"dynamic_result_management"` that contains
+  a constant `true` or `false` value of type `i1`
+
+These flags are attached as `llvm.module.flags` metadata to the module. They can
+be queried using the standard LLVM tools and follow the LLVM specification in
+behavior and purpose. Since module flags impact whether different modules can be
+merged and how, additional module flags may be added to the bitcode only if
+their behavior is set to `Warning`, `Append`, `AppendUnique`, `Max`, or `Min`.
+It is at the discretion of the maintainers for various components in the QIR
+stack to discard module flags that are not explicitly required or listed as
+optional flags in the QIR specification.
+
+### Specification Version
+
+The required flags `"qir_major_version"` and `"qir_minor_version"` identify the
+major and minor version of the specification that the QIR bitcode adheres to.
+
+- Since the QIR specification may introduce breaking changes when updating to a
+  new major version, the behavior of the `"qir_major_version"` flag must be set
+  to `Error`; merging two modules that adhere to different major versions must
+  fail.
+
+- The QIR specification is intended to be backward compatible within the same
+  major version, but may introduce additional features as part of newer minor
+  versions. The behavior of the `"qir_minor_version"` flag must hence be `Max`,
+  such that merging two modules compiled for different minor versions results in
+  a module that adheres to the newer of the two versions.
+
+### Memory Management
+
+Each bitcode file contains the information whether pointers of type `%Qubit*`
+and `%Result*` point to a valid memory location, or whether a pointer merely
+encodes an integer constant that identifies which qubit or result the value
+refers to. This information is represented in the form of the two module flags
+named `"dynamic_qubit_management"` and `"dynamic_result_management"`. Within the
+same bitcode module, there can never be a mixture of the two different kinds of
+pointers. The behavior of both module flags correspondingly must be set to
+`Error`. As detailed in the section on [qubit and result
+usage](#qubit-and-result-usage), a Base Profile compliant program must not make
+use of dynamic qubit or result management. The value of both module flags hence
+must be set to `false`.
```

Comment:
Can we just not add the flags to the base profile if they're not going to be used anyway?

Reply:
What do you mean by them not going to be used? 

To give some context on what my thinking was: 
The flags are useful to have within general QIR (i.e. QIR that has not yet been compiled for a specific target). QIR from different sources might be combined into a single program at this stage, and the flags are intended to make this fail when one uses dynamic management and the another does not. Since generally the expectation is that targeted QIR is a subset of general QIR, if flags are required in general, they also should be present (possibly with a different value of course) in the profile.

Reply:
Yeah that's fine, didn't realize they dealt with merging too.